dbos/_sys_db.py

```diff
@@ -1460,21 +1460,37 @@ def remove_from_queue(self, workflow_id: str, queue: "Queue") -> None:
                     .values(completed_at_epoch_ms=int(time.time() * 1000))
                 )
 
-    def clear_queue_assignment(self, workflow_id: str) -> None:
+
+    def clear_queue_assignment(self, workflow_id: str) -> bool:
         if self._debug_mode:
             raise Exception("called clear_queue_assignment in debug mode")
-        with self.engine.begin() as c:
-            c.execute(
-                sa.update(SystemSchema.workflow_queue)
-                .where(SystemSchema.workflow_queue.c.workflow_uuid == workflow_id)
-                .values(executor_id=None, started_at_epoch_ms=None)
-            )
-            c.execute(
-                sa.update(SystemSchema.workflow_status)
-                .where(SystemSchema.workflow_status.c.workflow_uuid == workflow_id)
-                .values(executor_id=None, status=WorkflowStatusString.ENQUEUED.value)
-            )
 
+        with self.engine.connect() as conn:
+            with conn.begin() as transaction:
+                conn.execute(sa.text("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ"))
```

Comment:
This doesn't need repeatable read. Also, repeatable read can cause unnecessary failures when two executors are updating the same row.